interface/admin.py

```diff
@@ -1,6 +1,48 @@
+from zipfile import ZipFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
 from django.contrib import admin
+from django.db.models import F
+from django.http import HttpResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, _assignment):
+        assignment = _assignment[0]
```

Comment:
So you're expecting a single assignment? You should check, and throw an error otherwise.